src/applications/proxy-rewrite/redirects/index.js

```diff
@@ -22,3 +22,19 @@ export default function redirectIfNecessary(currentWindow) {
     }`;
   }
 }
+
+function localRedirectIfNecessary() {
+  const target = new URLSearchParams(window.location.search).get('target');
+  const matchedRedirect = crossDomainRedirects.find(
+    redirect => `https://${redirect.domain}${redirect.src}` === target,
+  );
+
+  if (matchedRedirect) {
+    // eslint-disable-next-line no-param-reassign
+    window.location.href = `${environment.BASE_URL}${matchedRedirect.dest}`;
+  }
+}
```

Comment:
we shouldn't make functions that are only executed in tests like this. we should extract the units from `redirectIfNecessary` and test them separately. This is skipping logic that should be covered by the functional test?

Reply:
you have identified only the thinnest slice of the pie that compose the many issues with this pull request

Reply:
😹